src/rageshake/rageshake.js

```diff
@@ -241,20 +241,27 @@ class IndexedDBLogStore {
 
         // Returns: a string representing the concatenated logs for this ID.
         function fetchLogs(id) {
-            const o = db.transaction("logs", "readonly").objectStore("logs");
-            return selectQuery(o.index("id"), IDBKeyRange.only(id),
-            (cursor) => {
-                return {
-                    lines: cursor.value.lines,
-                    index: cursor.value.index,
+            const objectStore = db.transaction("logs", "readonly").objectStore("logs");
+
+            return new Promise((resolve, reject) => {
+                const query = objectStore.index("id").openCursor(IDBKeyRange.only(id), 'next');
```

Comment:
In case there are more than MAX_LOG_SIZE log bytes, this will return the head of it, rather than the tail, right? I'm wondering whether the tail would be more interesting, and therefore iterate in reverse order here?

Reply:
I did wonder that, but that's nontrivial because of indexeddb and compound indexes. I concluded that if a single log file is going over the limit, we're doing something wrong (eg. letting bluebird spam deprecation warnings).